tests/cypress/integration/publisher/018-read-only-user/00-verify-that-read-only-user-cannot-create-update-api.js

```diff
@@ -0,0 +1,276 @@
+
+        cy.get('#http-panel1bh-header').click();
+        cy.get('#certs-add-btn').get('[aria-disabled="true"]').should('exist');
+        cy.get('#panel1bh-header').click();
+        cy.get('#certificateEndpoint').get('[aria-disabled="true"]').should('exist');
+        cy.get('button').get('[aria-label="Delete"]').should('be.disabled');
+
+        cy.contains('button', 'Save').should('be.disabled');
+
+        //11. Localscopes
+        cy.get('#left-menu-itemLocalScopes').click();
+        cy.contains('a','Add New Local Scope').get('[aria-disabled="true"]').should('exist');
+        cy.get('table').get('tbody').get('[data-testid="MUIDataTableBodyRow-0"]').get('[data-testid="MuiDataTableBodyCell-4-0"]').get('[aria-label="Edit creatorscope"]').get('[aria-disabled="true"]').should('exist');
+        cy.get('table').get('tbody').get('[data-testid="MUIDataTableBodyRow-0"]').get('[data-testid="MuiDataTableBodyCell-4-0"]').contains('button','Delete').should('be.disabled');
+
+        //12. Policies should be checked. for now cannot due to a UI bug
+
+        //13. monetization ,lifecycle menus are not visible to observer
+        cy.get('[data-testid="left-menu-itemlifecycle"]').should('not.exist');
+        cy.get('[data-testid="left-menu-monetization"]').should('not.exist');
+
+        //14. Properties
+        cy.get('#left-menu-itemproperties').click();
+        cy.get('#add-new-property').should('be.disabled');
+        cy.get('table').get('tbody').get('tr').contains('td','property1').should('be.visible');
+        cy.get('table').get('tbody').get('tr').get('[aria-label="Edit property1"]').should('be.disabled');
+        cy.get('table').get('tbody').get('tr').get('[aria-label="Remove property1"]').should('be.disabled');
+        cy.get('[data-testid="save-api-properties-btn"]').should('be.disabled');
+
+        //15. Deployments
+        cy.get('#react-root').scrollTo('bottom');
+        cy.get('#left-menu-itemdeployments').click();
+        cy.contains('button','Deploy New Revision').should('be.disabled');
+        cy.contains('button','Restore').should('be.disabled');
+        cy.contains('button','Delete').should('be.disabled');
+        cy.get('#undeploy-btn').should('be.disabled');
+        cy.get('table').get('tbody').get('tr').find('td').eq(2).get('[aria-disabled="true"]').should('exist');
+        
+        //16. Header buttons should also be disabled
+        cy.get('#itest-id-deleteapi-icon-button').should('not.exist');
+        cy.get('#create-new-version-btn').should('not.exist');
+    });
+
+    after(function () {
+        cy.logoutFromPublisher();
+
+        // Test is done. Now delete the api
+        cy.loginToPublisher(carbonUsername, carbonPassword);
+        cy.deleteApi(apiName, apiVersion);
+
+        // delete observer user.
+        cy.visit(`${Utils.getAppOrigin()}/carbon/user/user-mgt.jsp`);
+        cy.deleteUser(readOnlyUser);
+    })
+    
+});
```

Comment:
Let's add a new line at the end of this file

Reply:
fixed